main.cpp

```diff
@@ -53,6 +54,23 @@ struct Piano
         return octaves;
     }
 
+    JUCE_LEAK_DETECTOR(Piano)
+};
+
+struct PianoWrapper
+{
+    Piano* adress = nullptr; //Initilaize Pointer to a Piano UDT to nullptr
+
+    //Constructor for the PianoWrapper taking  a Pointer to a Piano UDT as an argument
+    PianoWrapper(Piano* p) : 
+    adress(p)        //updating the Pointer named piano to the adress of the UDT type pointerwe have a variable in the constructor
+    {
+
+    }
+    ~PianoWrapper()
+    {
+        delete adress; //deleting the piano pointer when the destructor is called.
+    }
```

Comment:
`address`, not `adress`
- [ ] correct spelling matters when other people read your code!  revise